src/masternode-sync.h

```diff
@@ -42,38 +42,32 @@ class CMasternodeSync
 
     // Time when current masternode asset sync started
     int64_t nTimeAssetSyncStarted;
-
-    // Last time when we received some masternode asset ...
-    int64_t nTimeLastMasternodeList;
-    int64_t nTimeLastPaymentVote;
-    int64_t nTimeLastGovernanceItem;
+    // ... last bumped
+    int64_t nTimeLastBumped;
     // ... or failed
     int64_t nTimeLastFailure;
 
     // Keep track of current block index
     const CBlockIndex *pCurrentBlockIndex;
 
-    bool CheckNodeHeight(CNode* pnode, bool fDisconnectStuckNodes = false);
     void Fail();
     void ClearFulfilledRequests();
 
 public:
     CMasternodeSync() { Reset(); }
 
-    void AddedMasternodeList() { nTimeLastMasternodeList = GetTime(); }
-    void AddedPaymentVote() { nTimeLastPaymentVote = GetTime(); }
-    void AddedGovernanceItem() { nTimeLastGovernanceItem = GetTime(); };
 
     void SendGovernanceSyncRequest(CNode* pnode);
 
     bool IsFailed() { return nRequestedMasternodeAssets == MASTERNODE_SYNC_FAILED; }
-    bool IsBlockchainSynced(bool fBlockAccepted = false);
+    bool IsBlockchainSynced() { return nRequestedMasternodeAssets > MASTERNODE_SYNC_IBD; }
     bool IsMasternodeListSynced() { return nRequestedMasternodeAssets > MASTERNODE_SYNC_LIST; }
     bool IsWinnersListSynced() { return nRequestedMasternodeAssets > MASTERNODE_SYNC_MNW; }
     bool IsSynced() { return nRequestedMasternodeAssets == MASTERNODE_SYNC_FINISHED; }
 
     int GetAssetID() { return nRequestedMasternodeAssets; }
     int GetAttempt() { return nRequestedMasternodeAttempt; }
+    void BumpAssetLastTime(std::string strFuncName);
```

Comment:
Separate out the logging, please, and get the function name etc. automatically.
It's wrong to pass in the log message like this, but I can see why you do it.
Use __func__ to pick up the function name automatically.
There isn't a straightforward way to pick up the class name automatically,
but we can add a template which provides it if needed.
This functionality really belongs in the logging util library, I guess.